src/test/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateServiceUnitTest.java

```diff
@@ -22,24 +35,70 @@
 import org.springframework.web.client.RestTemplate;
 
 import java.io.InputStream;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertTrue;
 
 @RunWith(SpringRunner.class)
 @EnableAutoConfiguration(exclude = MessageCollectorAutoConfiguration.class)
 @SpringBootTest(classes = {CommunicationApp.class, SecurityBeanOverrideConfiguration.class})
 public class EmailTemplateServiceUnitTest {
 
-    @Autowired
+    private static final String TENANT_KEY = "TEST";
+    private static final String UPDATED_TEMPLATE_NAME = "updated template name";
+    private static final String UPDATED_SUBJECT_NAME = "updated subject name";
+    private static final String EMAIL_SPECIFICATION_PATH = "/config/tenants/TEST/communication/email-spec.yml";
+    private static final String CUSTOM_EMAIL_SPECIFICATION_PATH = "/config/tenants/TEST/communication/custom-email-spec.yml";
+    private static final String EMAIL_SPECIFICATION_PATH_PATTERN = "/config/tenants/{tenantName}/communication/email-spec.yml";
+    private static final String CUSTOM_EMAIL_SPECIFICATION_PATH_PATTERN = "/config/tenants/{tenantName}/communication/custom-email-spec.yml";
+    private static final String CUSTOM_EMAIL_TEMPLATES_PATH = "/config/tenants/TEST/communication/custom-emails/";
+
+    private TenantContextHolder tenantContextHolder;
+
+    private EmailSpecService emailSpecService;
+
     private EmailTemplateService subject;
 
+    private final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+
+    @Mock
+    private CommonConfigRepository commonConfigRepository;
+
+    @Mock
+    private ApplicationProperties applicationProperties;
+
     @MockBean
     private SmppService smppService;
 
     @MockBean
     private RestTemplate restTemplate;
+    @Autowired
+    private freemarker.template.Configuration freeMarkerConfiguration;
+
+
+    @Before
+    public void init() {
+        tenantContextHolder = mock(TenantContextHolder.class);
+        mockTenant(TENANT_KEY);
+
+        when(applicationProperties.getEmailSpecificationPathPattern()).thenReturn(EMAIL_SPECIFICATION_PATH_PATTERN);
+        when(applicationProperties.getCustomEmailSpecificationPathPattern()).thenReturn(CUSTOM_EMAIL_SPECIFICATION_PATH_PATTERN);
+
```

Comment:
We don't need to mock application properties. Just add these properties in application.yml in test folders.

Reply:
replaced